tron/utils/scribereader.py

```diff
@@ -90,146 +141,108 @@ def read_log_stream_for_action_run(
-            for line in stream:
-                if num_lines == max_lines:
-                    truncated_output = True
-                    break
-                # it's possible for jobs to run multiple times a day and have obscenely large amounts of output
-                # so we can't just truncate after seeing X number of lines for the run number in question - we
-                # need to count how many total lines we've seen and bail out early to preserve tron's uptime
-                num_lines += 1
-
-                try:
-                    payload = json.loads(line)
-                except json.decoder.JSONDecodeError:
-                    log.error(f"Unable to decode log line from stream ({stream_name}) for {action_run_id}: {line}")
-                    malformed_lines += 1
-                    continue
-
-                if (
-                    payload.get("tron_run_number") == int(run_num)
-                    and payload.get("component") == component
-                    and payload.get("message") is not None
-                    and payload.get("timestamp") is not None
-                    and payload.get("cluster") == paasta_cluster
-                ):
-                    output.append((payload["timestamp"], payload["message"]))
-        except StreamTailerSetupError:
-            return [
-                f"No data in stream {stream_name} - if this is the first time this action has run and you expected "
-                "output, please wait a couple minutes and refresh."
-            ]
-        except socket.timeout:
-            return [
-                f"Unable to connect to stream {stream_name} - if this is the first time this action has run and you "
-                "expected output, please wait a couple minutes and refresh."
-            ]
-        finally:
-            stream.close()
-
-    # XXX: for some reason, we're occasionally getting data out of order from scribereader - so we'll sort based on
-    # timestamp until we can figure out what's causing this.
-    output.sort(key=operator.itemgetter(0))
-    lines = [line for _, line in output]
-    malformed = [f"{malformed_lines} encountered while retrieving logs"] if malformed_lines else []
+
+    # yelp_clog S3LogsReader is a newer reader that is supposed to replace scribe readers eventually.
+    if use_s3_reader:
+        end_date = max_date.date() if max_date else today
+
+        log.debug("Using S3LogsReader to retrieve logs")
+        s3_reader = S3LogsReader(ecosystem).get_log_reader(log_name=stream_name, min_date=start_date, max_date=end_date)
+        paasta_logs.iterate_logs(s3_reader, max_lines)
+    else:
+        end_date = max_date.date() if max_date else None  # type: ignore
+        use_tailer = today in {start_date, end_date}
+        use_reader = start_date != today and end_date is not None
+
+        host, port = get_scribereader_host_and_port(ecosystem, superregion, region)  # type: ignore  # the None case is covered by the check above
```

Comment:
we should be able to remove this `type: ignore` by fixing the return typing of `get_scribereader_host_and_port` - generally, `type: ignore` should be a last resort :)